include/nil/crypto3/hash/detail/tiger/basic_tiger_policy.hpp

```diff
@@ -0,0 +1,322 @@
+//---------------------------------------------------------------------------//
+// Copyright (c) 2018-2020 Mikhail Komarov <nemo@nil.foundation>
+// Copyright (c) 2020 Pavel Kharitonov <ipavrus@nil.foundation>
+//
+// Distributed under the Boost Software License, Version 1.0
+// See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt
+//---------------------------------------------------------------------------//
+
+#ifndef CRYPTO3_BASIC_TIGER_POLICY_HPP
+#define CRYPTO3_BASIC_TIGER_POLICY_HPP
+
+#include <nil/crypto3/detail/basic_functions.hpp>
+
+namespace nil {
+    namespace crypto3 {
+        namespace hash {
+            namespace detail {
+                template<std::size_t DigestBits>
+                struct basic_tiger_policy : public ::nil::crypto3::detail::basic_functions<64> {
+                    typedef ::nil::crypto3::detail::basic_functions<64> policy_type;
+
+                    constexpr static const std::size_t word_bits = policy_type::word_bits;
+                    typedef typename policy_type::word_type word_type;
+
+                    constexpr static const std::size_t block_bits = 512;
+                    constexpr static const std::size_t block_words = block_bits / word_bits;
+                    typedef std::array<word_type, block_words> block_type;
+
+                    constexpr static const std::size_t state_bits = DigestBits;
+                    constexpr static const std::size_t state_words = state_bits / word_bits;
+                    typedef std::array<word_type, state_words> state_type;
+
+                    constexpr static const std::size_t substitution_words = 256;
+                    typedef std::array<word_type, substitution_words> substitution_type;
+
+                    constexpr static const substitution_type sbox1 = {
```

Comment:
Just thinking out loud here. Has anyone considered making this compatible (or just testing, if it is compatible) with C++ standards lower, than C++17? I mean ```constexpr inline``` variables, as they are being used in here, were introduced only in C++17, as far as I remember (https://en.cppreference.com/w/cpp/language/inline). Earlier standards require different specifications. We have ```CONSTEXPR_INLINE_VARIABLE``` for this.